File: Data Structures & Algorithms/sliding-window-maximum/submission-0.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        res = []
        l, r = 0, k

        while r < len(nums):
            res.append(max(nums[l:r]))
            l+=1
            r+=1
        
        res.append(max(nums[l:r]))
        return res

File: Data Structures & Algorithms/subsets/submission-0.cpp
class Solution {
   public:
    vector<vector<int>> res;
    vector<vector<int>> subsets(vector<int>& nums) {
        backtrack(nums, 0, {});
        return res;
    }
    void backtrack(vector<int>& nums, int i, vector<int> subset) {
        if (i >= nums.size()) {
            res.push_back(subset);
            return;
        }
        subset.push_back(nums[i]);
        backtrack(nums, i + 1, subset);
        subset.pop_back();
        backtrack(nums, i + 1, subset);
    }
};
